core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -58,12 +65,16 @@
  *
  * Note: The HadoopCatalog requires that the underlying file system supports atomic rename.
  */
-public class HadoopCatalog extends BaseMetastoreCatalog implements Closeable {
+public class HadoopCatalog extends BaseMetastoreCatalog implements Closeable, SupportsNamespaces {
+  private static final Logger LOG = LoggerFactory.getLogger(HadoopCatalog.class);
+
   private static final String ICEBERG_HADOOP_WAREHOUSE_BASE = "iceberg/warehouse";
   private static final String TABLE_METADATA_FILE_EXTENSION = ".metadata.json";
+  private static final Joiner SLASH = Joiner.on("/");
   private static final PathFilter TABLE_FILTER = path -> path.getName().endsWith(TABLE_METADATA_FILE_EXTENSION);
   private final Configuration conf;
   private String warehouseLocation;
+  private HadoopFileIO defaultFileIo = null;
```

Comment:
I don't think this variable is used. If the warehouse location is used for all namespaces, then this could be initialized in the constructor and used instead of calling `Util.getFs` in multiple methods.